backend/src/test/java/com/pickpick/auth/application/AuthServiceTest.java

```diff
@@ -0,0 +1,74 @@
+package com.pickpick.auth.application;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.pickpick.member.domain.Member;
+import com.pickpick.member.domain.MemberRepository;
+import com.slack.api.methods.MethodsClient;
+import com.slack.api.methods.SlackApiException;
+import com.slack.api.methods.request.oauth.OAuthV2AccessRequest;
+import com.slack.api.methods.request.users.UsersIdentityRequest;
+import com.slack.api.methods.response.oauth.OAuthV2AccessResponse;
+import com.slack.api.methods.response.oauth.OAuthV2AccessResponse.AuthedUser;
+import com.slack.api.methods.response.users.UsersIdentityResponse;
+import com.slack.api.methods.response.users.UsersIdentityResponse.User;
+import java.io.IOException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.transaction.annotation.Transactional;
+
+@AutoConfigureMockMvc
+@Transactional
+@SpringBootTest
+class AuthServiceTest {
+
+    @Autowired
+    private AuthService authService;
+
+    @MockBean
+    private MethodsClient slackClient;
+
+    @Autowired
+    private MemberRepository members;
+
+    @DisplayName("로그인")
+    @Test
+    void login() throws SlackApiException, IOException {
+        // given
+        Member member = new Member("slackId", "username", "thumbnail.png");
+        members.save(member);
+
+        when(slackClient.oauthV2Access(any(OAuthV2AccessRequest.class)))
+                .thenReturn(generateOAuthV2AccessResponse());
+        when(slackClient.usersIdentity(any(UsersIdentityRequest.class)))
+                .thenReturn(generateUsersIdentityResponse(member.getSlackId()));
```

Comment:
```suggestion
        given(slackClient.oauthV2Access(any(OAuthV2AccessRequest.class)))
                .willReturn(generateOAuthV2AccessResponse());
        given(slackClient.usersIdentity(any(UsersIdentityRequest.class)))
                .willReturn(generateUsersIdentityResponse(MEMBER_SLACK_ID));
```

given에 해당하는 조건문 설정은 given, willReturn을 사용해보면 어떨까요? 🍪

Reply:
무슨 차이인지 몰라서 간단히 검색한 결과를 함께 공유드립니다
- thenReturn -> Mockito 사용.
- willReturn -> BDDMockito 사용. 

### BDDMockito
- = Behavior-Driven Development; 행위 주도 개발
- Mockito을 상속한 클래스
- 동작 방식이나 사용 방법도 Mockito와 유사
- Mockito에서 신경쓰이는 네이밍을 의도에 맞게 변경해서 사용 가능 (ex: 프로젝트 코드에서 `//given` 어노테이션 밑에 `when()` 메서드가 사용되고 있는데 BDDMockito를 사용하면 `given()` 메서드를 사용할 수 있게 됨)

> 참고자료: https://velog.io/@lxxjn0/Mockito%EC%99%80-BDDMockito%EB%8A%94-%EB%AD%90%EA%B0%80-%EB%8B%A4%EB%A5%BC%EA%B9%8C

리차드 의견에 동의합니다 새로운 정보 감사합니다😄